server/src/workflows/themes/postThemes/postThemes.ts

```diff
@@ -19,8 +18,10 @@ export const addThemeInNotificationSettingsForUser = (theme: Theme) => (user: Ap
     themes: { ...user.notificationsSettings.themes, [`${theme._id}`]: hasOneNotificationEnabled(user) },
   });
 
-const updateUsersNotificationsSettings = async (theme: Theme) =>
-  getAllAppUsers().then(map(addThemeInNotificationSettingsForUser(theme)));
+const updateUsersNotificationsSettings = async (theme: Theme) => {
+  const appUsers = await getAllAppUsers()
+  return Promise.all(appUsers.map(addThemeInNotificationSettingsForUser(theme)));
```

Comment:
not sure if relevant but maybe an issue here if you have many users : this will cause many parallel calls to `updateNotificationsSettings`.

Reply:
Great catch @revolunet !  I'm creating a new ticket for this because it's beyond the scope of the work here.  This code is called when an admin user created a new "theme".  Themes are categories such as "Health" or "Learn a trade".  These are fairly stable in the app, however since that call does an update on all users, it should be batched.  Also depending on the expected latency, the rest endpoint would probably need to be modified since it's currently waiting for all processing to take place and may timeout as the number of user's scale.  I'll also add a todo.

<img width="1472" alt="Screenshot 2024-07-08 at 21 01 46" src="https://github.com/refugies-info/karfur/assets/67010/a00dac4e-1404-4c53-8495-ec1d332b02f1">

Reply:
See #2166